### PR TITLE
Added more logging options to configuration

### PR DESCRIPTION
**Pull request by MarioIvancik**
https://youtrack.bringauto.com/issue/BAF-991/More-logging-options-in-configuration

## Release Notes

- **New Features**
  - Introduced a new logging configuration section, enhancing control over console and file logging.
  - Updated the version number of the project to 3.3.0.

- **Documentation**
  - Improved the README.md for clarity, adding a dedicated logging section and updating command-line arguments.
  
- **Bug Fixes**
  - Streamlined logging settings in configuration files, removing outdated parameters for a cleaner structure.
  
- **Chores**
  - Refactored logging settings management for better organization and modularity.

### CODE REVIEW

File: source/bringauto/common_utils/EnumUtils.cpp [hunk 1]
```diff
@@ -75,6 +76,24 @@ settings::FleetProvider EnumUtils::valueToEnum(std::string toEnum) {
 	return settings::FleetProvider::E_INVALID;
 }
 
+template <>
+logging::LoggerVerbosity EnumUtils::valueToEnum(std::string toEnum) {
```
Comment (koudis) [inline source/bringauto/common_utils/EnumUtils.cpp:80]: Do we have unit tests for these functions? + settings parser?

Comment (MarioIvancik) [reply source/bringauto/common_utils/EnumUtils.cpp:80]: Unit tests are now added

File: resources/config/default.json [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
+	"logging": {
+		"console": {
+			"level": "DEBUG",
+			"use": true
+		},
+		"file": {
+			"level": "DEBUG",
+			"use": false,
+			"path": ""
+		}
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply resources/config/default.json:14]: invalid file paths are being handled

File: test/source/SettingsParserTest.cpp [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+TEST_F(SettingsParserTest, DuplicateCmdOption){
+	bool failed = false;
+	try	{
+		auto result = parseConfig(ConfigMock::Config(), {"--config=/path"});
+	}catch (std::invalid_argument &e){
+		EXPECT_STREQ(e.what(), "Cmd arguments are not correct.");
+		failed = true;
+	}
+	EXPECT_TRUE(failed);
+}
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply test/source/SettingsParserTest.cpp:268]: the config param is added by default so providing another one as a parseConfig parameter causes there to be two

File: test/source/SettingsParserTest.cpp [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+
+/**
+ * @brief Test if duplicate command line options are correctly handled
+ */
+TEST_F(SettingsParserTest, DuplicateCmdOption){
+	bool failed = false;
+	try	{
+		auto result = parseConfig(ConfigMock::Config(), {"--config=/path"});
+	}catch (std::invalid_argument &e){
+		EXPECT_STREQ(e.what(), "Cmd arguments are not correct.");
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply test/source/SettingsParserTest.cpp:264]: The unit tests are passing so this statement is false

File: test/source/EnumUtilsTest.cpp [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	EXPECT_EQ(bacu::EnumUtils::valueToEnum<bas::FleetProvider>(std::string("INTERNAL-PROTOCOL")), bas::FleetProvider::E_INTERNAL_PROTOCOL);
+	EXPECT_EQ(bacu::EnumUtils::valueToEnum<bas::FleetProvider>(std::string("NO-CONNECTION")), bas::FleetProvider::E_NO_CONNECTION);
+	EXPECT_EQ(bacu::EnumUtils::valueToEnum<bas::FleetProvider>(std::string("INVALID")), bas::FleetProvider::E_INVALID);
+
+	EXPECT_EQ(bacu::EnumUtils::valueToEnum<balog::LoggerVerbosity>(std::string("DEBUG")), balog::LoggerVerbosity::Debug);
+	EXPECT_EQ(bacu::EnumUtils::valueToEnum<balog::LoggerVerbosity>(std::string("INFO")), balog::LoggerVerbosity::Info);
+	EXPECT_EQ(bacu::EnumUtils::valueToEnum<balog::LoggerVerbosity>(std::string("WARNING")), balog::LoggerVerbosity::Warning);
+	EXPECT_EQ(bacu::EnumUtils::valueToEnum<balog::LoggerVerbosity>(std::string("ERROR")), balog::LoggerVerbosity::Error);
+	EXPECT_EQ(bacu::EnumUtils::valueToEnum<balog::LoggerVerbosity>(std::string("CRITICAL")), balog::LoggerVerbosity::Critical);
+	EXPECT_EQ(bacu::EnumUtils::valueToEnum<balog::LoggerVerbosity>(std::string("INVALID")), balog::LoggerVerbosity::Debug);
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply test/source/EnumUtilsTest.cpp:53]: There is a warning that mentions any invalid verbosity defaulting to debug level

**Review by koudis — APPROVED** [review]

**Pull request merged**